examples/asr/speech_to_text_infer.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This script serves three goals:
+    (1) Demonstrate how to use NeMo Models outside of PytorchLightning
+    (2) Shows example of batch ASR inference
+    (3) Serves as CI test for pre-trained checkpoint
+"""
+
+from argparse import ArgumentParser
+
+import torch
+
+from nemo.collections.asr.metrics.wer import WER, word_error_rate
+from nemo.collections.asr.models import EncDecCTCModel
+from nemo.utils import logging
+
+can_gpu = torch.cuda.is_available()
+
+
+def main():
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--asr_model", type=str, default="QuartzNet15x5Base-En", required=True, help="Pass: 'QuartzNet15x5Base-En'",
+    )
+    parser.add_argument("--dataset", type=str, required=True, help="path to evaluation data")
+    parser.add_argument("--wer_target", type=float, default=None, help="used by test")
+    parser.add_argument("--batch_size", type=int, default=4)
+    parser.add_argument("--wer_tolerance", type=float, default=1.0, help="used by test")
+    parser.add_argument(
+        "--normalize_text", default=True, type=bool, help="Normalize transcripts or not. Set to False for non-English."
+    )
+    args = parser.parse_args()
+
+    if args.asr_model.endswith('.nemo'):
+        logging.info(f"Using local ASR model from {args.asr_model}")
+        asr_model = EncDecCTCModel.restore_from(restore_path=args.asr_model)
+    else:
+        logging.info(f"Using NGC cloud ASR model {args.asr_model}")
+        asr_model = EncDecCTCModel.from_pretrained(model_name=args.asr_model)
+    asr_model.setup_test_data(
+        test_data_config={
+            'sample_rate': 16000,
+            'manifest_filepath': args.dataset,
+            'labels': asr_model.decoder.vocabulary,
+            'batch_size': args.batch_size,
+            'normalize_transcripts': args.normalize_text,
+        }
+    )
+    if can_gpu:
+        asr_model = asr_model.cuda()
+    asr_model.eval()
+    labels_map = dict([(i, asr_model.decoder.vocabulary[i]) for i in range(len(asr_model.decoder.vocabulary))])
+    wer = WER(vocabulary=asr_model.decoder.vocabulary)
+    hypotheses = []
```

Comment:
Is there a particular reason we are doing this manually and not depending on trainer.test()?

Reply:
yes: for inference wanted to have PTL-free version. IMHO, nobody will actually run inference with PTL for anything other than publication, so wanted to give "pure" torch variant

Reply:
Good point